### ISSUE TITLE
v2 cmake presets: Generates files which are incompatible with installed CMake ` Unrecognized "version" field`

### ISSUE THREAD

**Issue by prince-chrismc** [conan-io/conan#13520]
UX gripe, the output says CMakePresets are supported above version 3.19 however the `"include"` which as used were introduced in version 4 (cmake v3.23) https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html

The result is a confusing message that takes a while to sort out 😕 

The code comments clearly forces it to version 4 so conan should spit out a higher when working in v4 mode?

More instructions about how to lower the cmake requirement and use presets? 

```
C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>conan-script source . --version 9.1.0
conanfile.py (fmt/9.1.0): Calling source() in C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all\src
Downloading fmt-9.1.0.zip

conanfile.py (fmt/9.1.0): Unzipping 4.5MB, this can take a while
Unzipping 100 %

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>conan-script install . --version 9.1.0

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
Graph root
    conanfile.py (fmt/9.1.0): C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all\conanfile.py

======== Computing necessary packages ========

======== Installing packages ========

======== Finalizing install (deploy, generators) ========
conanfile.py (fmt/9.1.0): Calling generate()
conanfile.py (fmt/9.1.0): Generators folder: C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all\build\generators
conanfile.py (fmt/9.1.0): CMakeToolchain generated: conan_toolchain.cmake
conanfile.py (fmt/9.1.0): Preset 'conan-default' added to CMakePresets.json. Invoke it manually using 'cmake --preset conan-default'
conanfile.py (fmt/9.1.0): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE=C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all\build\generators\conan_toolchain.cmake -DFMT_DOC=OFF -DFMT_TEST=OFF -DFMT_INSTALL=ON -DFMT_LIB_DIR=lib -DFMT_OS=ON -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
conanfile.py (fmt/9.1.0): CMakeToolchain generated: CMakePresets.json
conanfile.py (fmt/9.1.0): CMakeToolchain generated: ..\..\src\CMakeUserPresets.json
conanfile.py (fmt/9.1.0): Generating aggregated env files
conanfile.py (fmt/9.1.0): Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
Install finished succesfully

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src  --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: Unrecognized "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: Unrecognized "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: No "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: Unrecognized "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: Unrecognized "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: Invalid "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake src --preset conan-default
CMake Error: Could not read presets from C:/Users/Chris McArthur/Documents/conan-center-index/recipes/fmt/all/src: Unrecognized "version" field

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>conan-script --version
Conan version 2.0.2

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\fmt\all>cmake --version
cmake version 3.22.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

https://github.com/conan-io/conan/blob/f0a67283508daa6e62eb123eee36d8e6f87d19b1/conan/tools/cmake/presets.py#L231-L238

**Comment by ahmedbodi** [comment]
whats the fix for this?

**Comment by memsharded** [comment]
Hi @ahmedbodi 

The fix is to use CMake >= 3.23, the above is mostly a suggestion for UX improvement with the error message. But to use the Conan genererated presets CMake>=3.23 is necessary, please upgrade your CMake and try again.

**Comment by memsharded** [comment]
Conan 2 has added the message:

```
 CMakeToolchain: Preset '{name}' added to CMakePresets.json.
                    (cmake>=3.23) cmake --preset {name}
                    (cmake<3.23) cmake <path> -G
```
In a different color, to highlight it a bit, in the output when using this generator. 

As checks are not possible in the ``CMakePresets.json`` itself, this looks the best Conan can do at the moment. Thanks for the feedback!
